### ISSUE TITLE
[M12] Full OCM lifetime evaluation across language, work, science, transfer, continual learning, and self-reorganization

### ISSUE THREAD

**Issue by SzeChunYiu** [SzeChunYiu/ORION-OCM#14]
# M12 — Full heterogeneous OCM lifetime evaluation

**Parent:** #1  
**LOCKED until:** #13 closes and all prior required gates have completed  
**Admissible terminal:** `FULL_OCM_RESIDUAL_SUPPORTED`, `PARENT_SUFFICIENT`, or `CANNOT_CHECK`

## Goal

Evaluate **one long-lived OCM instance** as a Machine Epistemics application across heterogeneous tasks and changing environments.

M12 is not a collection of independent leaderboard runs. The machine must carry a persistent knowledge/experience state through the lifetime, learn new capabilities, transfer only where justified, correct itself when evidence changes, and—where M11 is supported—propose governed cognitive repairs.

Central question:

```text
Does a machine built around explicit, revisable epistemic state and lifelong operator/representation learning
produce a measurable residual over the strongest faithful parent system
when both receive matched information, tools, memory, interaction, and resource accounting?
```

No global “intelligence score” or universal optimality claim is implied.

---

## 1. Freeze the evaluation ecology

Before protected lifetime execution, commit:

```text
task domains
sequence/order distributions
information channels
external tools
authority/safety policy
learning permissions
memory/storage limits
compute/time budgets
interaction budgets
protected task generators/data
comparators
primary metrics
statistical analysis
stopping rules
claim tiers
```

Use multiple pre-registered lifetime orderings/seeds to prevent one favorable curriculum from defining the result.

---

## 2. One persistent machine

The principal OCM arm must use one stateful runtime identity across the lifetime.

Required persistence across phases:

```text
knowledge/evidence
language learning
learned operators/skills
transfer maps
failure/counterexample knowledge
representation/topology state
learning-policy state if enabled
self-change lineage if adopted
```

Task-specific caches may expire, but the machine may not be reset between domains unless a declared experimental arm tests reset behavior.

A set of separately trained specialist systems does not count as the main OCM lifetime arm.

---

## 3. Lifetime phases

Freeze a sequence with at least:

### Phase A — language/social communication

- normal multi-turn conversation;
- new vocabulary/construction learning;
- correction/ambiguity;
- bounded knowledge questions;
- unfamiliar linguistic convention.

### Phase B — enterprise/knowledge work

- information retrieval/verification;
- policy-constrained task;
- multi-turn clarification;
- tool/action workflow;
- evidence-grounded report.

### Phase C — software/technical work

- diagnose/fix a fault;
- environment drift/new tool version;
- verify/rollback;
- transfer general work methods.

### Phase D — data/scientific work

- analyze evidence/data;
- choose an informative experiment/analysis;
- revise a hypothesis;
- scientific coding/formal checker task;
- report uncertainty.

### Phase E — novel cross-domain tasks

Tasks deliberately require partial combinations of structures learned in A–D, with both beneficial and harmful-looking analogies.

### Phase F — revision event

Revoke/correct important earlier evidence and require dependent conclusions/skills to update while unrelated capabilities remain.

### Phase G — cognitive failure / self-repair

If M11 is supported, introduce a protected failure whose layer is not disclosed and test diagnosis/proposal/shadow assurance/adoption. If M11 was `PARENT_SUFFICIENT`, run the strongest parent repair mechanism instead and preserve that terminal.

---

## 4. Fresh protected tasks

Public benchmarks are useful but contamination and overfitting risk make them insufficient for the primary lifetime claim.

Create fresh protected tasks with:

- exact graders/oracle state where possible;
- human expert review for realism;
- generated variants after system freeze;
- no publication of solutions before evaluation;
- versioned/hash-bound task manifests.

Use public benchmarks as secondary comparability anchors, refreshing versions at execution time. Candidate anchors include current language, enterprise, terminal/computer-use, scientific coding/research and formal-math benchmarks from M7–M10.

---

## 5. Transfer challenge across domains

Pre-register cross-domain transfer cases such as:

```text
clarification strategy: language → enterprise task
verify-after-action: work → software → science
fault isolation: software → experiment troubleshooting
uncertainty reporting: science → work decision
hierarchical decomposition: language planning → technical task planning
```

Also include deceptive analogies where transfer must be refused.

Track exactly what object/operator/adapter was reused and whether it causally improved outcome.

---

## 6. Lifelong learning challenge

Introduce genuinely new capabilities after initial freeze:

- new language construction;
- new enterprise policy/workflow;
- new software tool/API;
- new analysis method;
- new domain mapping/representation.

For each measure:

```text
information/examples required
interactions/tool calls
update compute/time
immediate performance
future reuse
retention
negative transfer
local correction/revocation
```

Compare against matched parent update mechanisms: in-context memory, retrieval, fine-tuning/adapters, explicit skill library, retraining, etc.

---

## 7. Nonstationarity / revision

Change the world during the lifetime:

- policy version changes;
- factual source retracted/corrected;
- software API changes;
- dialogue convention changes;
- scientific evidence invalidated;
- tool unavailable/replaced.

Measure:

- detection delay;
- stale behavior rate;
- affected-dependency reopening;
- adaptation cost;
- unnecessary regression elsewhere.

OCM must not equate prior success with eternal validity.

---

## 8. Unknown/frontier behavior

Include tasks outside current competence/knowledge.

Measure whether the system chooses appropriately among:

```text
answer
clarify
retrieve/search if allowed
experiment if allowed
learn
compose
report unknown/uncertain
propose representation/method change
CANNOT_CHECK
```

A system that always attempts an answer is penalized separately from task accuracy.

---

## 9. Strongest faithful whole-system parent

Construct the strongest feasible parent product at execution time, not a deliberately simple baseline.

Parent may include:

```text
current frontier foundation model/agent
same external tools
same task documents/knowledge corpus
retrieval/vector/episodic memory
long-term skill/trajectory library
post-deployment fine-tuning/adapters if OCM may learn
planner/router
verification/checkers
persistent user/task memory
self-reflection/self-repair mechanism if allowed
```

Every component/power OCM uses must either be given to the comparator or identified as the experimental difference.

If no credible matched parent can be built for a claim, mark that claim `CANNOT_CHECK_MATCHED_PARENT`.

---

## 10. External frontier reference

Separately evaluate current best-available commercial/open agent systems on public/compatible parts of the suite.

Purpose:

- calibrate absolute usefulness/product quality;
- identify practical gaps;
- prevent OCM from appearing impressive only against small matched models.

These models may have far more pretraining exposure, so they are **reference arms**, not automatically matched causal comparators.

Refresh named models immediately before the frozen evaluation; do not hard-code 2026 model names into the scientific claim.

---

## 11. Claim tiers

Predefine claim tiers so one positive metric cannot be inflated into “AGI.”

### Tier 0 — operational

One persistent OCM completes the full evaluation without integrity failure.

### Tier 1 — capability

OCM reaches a frozen absolute quality threshold on one/more domains.

### Tier 2 — lifelong-learning residual

OCM learns/corrects later capabilities with a pre-registered advantage over strongest matched parent.

### Tier 3 — cross-domain transfer residual

A pre-registered beneficial partial-transfer advantage survives harmful-transfer controls.

### Tier 4 — epistemic-integrity residual

Explicit warrant/revocation/unknown machinery improves correction/calibration or reduces unsupported commitment under matched resources.

### Tier 5 — self-reorganization residual

Governed self-change adds an independently verified residual beyond strongest parent repair/meta-learning mechanism.

### Tier 6 — broad OCM residual

A joint pre-registered vector/residual remains across heterogeneous lifetime tasks without unacceptable regression/resource trade.

No tier by itself authorizes claims of consciousness, human equivalence, or universal intelligence.

---

### Task/product

```text
success/correctness
human usefulness
multi-turn/workflow completion
scientific/formal validity
```

### Learning

```text
competence vs cumulative information
post-deployment acquisition cost
retention/forgetting
future-task acquisition cost
```

### Transfer

```text
positive gain
negative harm
transfer precision
adapter cost
```

### Epistemics

```text
unsupported commitment rate
calibration
honest-unknown accuracy
source/provenance coverage
revocation/correction correctness
CANNOT_CHECK integrity
```

### Self-repair

```text
diagnosis accuracy
minimum-change accuracy
preservation
rollback
```

### Resources

```text
training/update compute
inference compute/latency
memory/storage/index size
external IO/tool calls
interaction burden
verification cost
self-improvement search cost
```

Report the vector, Pareto fronts and task-ecology-specific tradeoffs; do not collapse everything into a single global score.

---

## 13. Lifetime learning curves

Plot/record competence against:

```text
cumulative words/tokens
cumulative labeled/aligned examples
interaction turns
tool/experiment calls
compute
wall time
memory growth
```

Key question: does useful prior structure make later learning cheaper?

Compare slopes/areas under matched information regimes.

---

## 14. Integrity kill gates

Any of the following invalidates the corresponding run/claim until resolved:

- protected answer/test leakage;
- comparator information mismatch mislabeled as matched;
- hidden external LLM supplies mechanism-arm answers;
- revoked evidence remains externally committed as live;
- self-change edits/chooses its protected evaluator;
- external authority bypass;
- task outcomes missing from report;
- resource cost deliberately excluded;
- benchmark/version cannot be reproduced;
- `CANNOT_CHECK` silently treated as PASS.

Integrity failure is not averaged away by high task performance.

---

## 15. Red-team / hostile campaign

Independent or separately authored hostiles should target:

```text
language answer laundering
memory contamination
false transfer
stale knowledge
policy conflicts
tool side effects
resource exhaustion
benchmark gaming
reward hacking
self-evaluator modification
false novelty attribution
```

Require adversarial tasks unknown to the component implementers where feasible.

---

## 16. Statistical/reporting protocol

Before evaluation freeze:

- define sample sizes/power where possible;
- paired tests/bootstrap/permutation methods;
- confidence intervals;
- multiple-comparison handling;
- human-rating model/aggregation;
- failure/exclusion criteria;
- seed/repetition policy.

Publish all frozen protected outcomes, including failures and parent wins.

No selective task-family reporting.

---

## 17. Replication

Require at least one independent clean-environment replay of the principal terminal before a broad residual claim.

Replication should use:

```text
same frozen code/manifest
fresh machine/environment
content hashes
same comparator protocol
independent result receipt
```

For expensive tasks where exact full replication is infeasible, predefine the replication subset before outcomes.

---

## 18. ORION-V2 feedback package

At completion, emit Machine Epistemics research questions back to ORION-V2 for every unresolved failure family.

Each package includes:

```text
observed OCM failure
trace/evidence
strongest parent behavior
suspected epistemic mechanism
candidate research question
why application-level engineering was insufficient
```

This preserves the research↔application loop rather than turning ORION-OCM back into an unconstrained theory repo.

---

## 19. Deliverables

- [ ] frozen heterogeneous lifetime protocol
- [ ] fresh protected task generators/sets
- [ ] public benchmark/version manifest
- [ ] one persistent OCM release candidate
- [ ] strongest whole-system parent
- [ ] frontier reference arms
- [ ] information/resource receipts
- [ ] transfer and continual-learning challenge
- [ ] nonstationarity/revocation challenge
- [ ] unknown/frontier behavior challenge
- [ ] self-repair challenge if applicable
- [ ] red-team/hostile campaign
- [ ] blinded human evaluation where required
- [ ] statistical report
- [ ] independent replication receipt
- [ ] claim-tier matrix
- [ ] ORION-V2 research feedback packet
- [ ] final OCM technical report

---

## Exit gate

Close with one of:

```text
FULL_OCM_RESIDUAL_SUPPORTED
PARENT_SUFFICIENT
CANNOT_CHECK
```

`FULL_OCM_RESIDUAL_SUPPORTED` is authorized only if the pre-registered broad residual survives the strongest faithful parent, integrity gates, resource accounting, negative-transfer/correction stress, and independent replay.

`PARENT_SUFFICIENT` is a successful scientific outcome if existing parent systems explain the result; it should trigger simplification and new Machine Epistemics research questions rather than narrative inflation.

M12 is the end of this roadmap version, not the end of Machine Epistemics.

**Comment by SzeChunYiu** [comment]
M12 delivered in #27 (pre-registration V1 → DEV_CALIBRATION after first outcome access, V2 frozen and PROTECTED; independent replication MATCH on a second host).

**Final terminal: `FULL_OCM_RESIDUAL_SUPPORTED`** — in scope: relative to the matched whole-system parent buildable in this environment (M7 matched parent + M9 skill library + M10 parent procedures + parameter search/retry, identical information and budgets). Against a frontier foundation-model whole-system parent (§9) the claim is `CANNOT_CHECK_MATCHED_PARENT`; human usefulness rating and external benchmarks are CANNOT_CHECK.

Claim tiers 0–6 hold under the pre-registered rules; exactly one family is inferential (A conversations, 53/54 vs 33/54, n = 54, RESIDUAL_A at δ = 0.05); lifelong-learning (later domain 7 vs 12), transfer (precision 1.0 vs 2 harmful accepted), epistemic integrity (stale 0 vs 1; reopened 3/3; intact 2/2), and self-repair (3/3 vs 0/3 on undisclosed faults) are descriptive at one lifetime. Kill gates all zero (incl. the new identity reset/split gate). Deliverables §19: frozen protocol ✔, fresh protected generators ✔, benchmark manifest (CANNOT_CHECK, recorded) ✔, one persistent release candidate ✔, whole-system parent ✔, frontier arms ✗ (CANNOT_CHECK recorded), information/resource receipts ✔, transfer/continual challenge ✔, revocation challenge ✔, unknown/frontier challenge ✔, self-repair challenge ✔, red-team hostiles (M7 laundering, p-hack, lookalike, self-approve/evaluator-edit mutants) ✔, blinded human evaluation ✗ (CANNOT_CHECK), statistical report ✔, replication receipt ✔, claim-tier matrix ✔, ORION-V2 feedback packet ✔ (P1–P6), technical report ✔ (`docs/M12_LIFETIME_REPORT.md`).

Self-application rows S31–S33 record the three protocol defects the first outcome access exposed (harness identity split; pseudo-replication; mislabelled planted fault); none was a mechanism change.

**Comment by SzeChunYiu** [comment]
M12 V3 (PR #32): eight paired lifetimes on per-lifetime protected streams — the lifetime as the unit of inference (theory batch 6 F2). Pre-registered sign test over 8 lifetime differences: 10 families reject in OCM's favour (8/8, p = 0.0078: conversations, factual, honest unknown, lessons, negative transfer, causal identification, communication gate, cross-domain transfer, revision integrity, self-repair); 6 families are ties (work B/C, experiment selection, shared analysis, kernel, unknown-domain refusal). Kill gates 0 (incl. ledger-chain identity). Replication on a second host: MATCH. Decision: OCM_LIFETIME_RESIDUAL_SUPPORTED in scope (matched whole-system parent buildable here; frontier-class parent CANNOT_CHECK_MATCHED_PARENT). Reference arm (local open-weight LLM, F8, never in the decision) on the same streams: factual 228/240, honest unknown 0/160, lessons 24/48. Report: docs/M12_V3_PAIRED_LIFETIMES_REPORT.md.

**Comment by SzeChunYiu** [comment]
M12 V4 (PR #34, theory batch 7 intake): fresh streams with a world-true out-of-scope half, one pre-registered primary family (A conversations) under the one-sided ≥7/8 sign rule (size 9/256, power 0.81), six secondaries at α/6, collapsed-one-coin flag. Primary: 8/8, p = 0.0039, not collapsed → OCM_LIFETIME_RESIDUAL_SUPPORTED (replication MATCH). Secondaries all reject but are flagged collapsed (identical differences across streams). Reference arm four-class: licensed 233, unlicensed-true 199, unlicensed-false 38 of 480 — the pretraining channel made visible by the world-true half. Ledger S37 (V3 rule under-specified), S38 (whale/mammal label defect). Report: docs/M12_V4_PAIRED_LIFETIMES_REPORT.md.

**Comment by SzeChunYiu** [comment]
M12 V4-R (re-evaluation of the frozen V4 streams under the corrected adoption gate from #39, on the current runtime; note docs/M12_V4R_REEVALUATION_NOTE.md, result research/ocm-m12/M12_PAIRED_LIFETIMES_EVAL_V4R.json): primary family A conversations OCM 0.9954 vs parent 0.6065, 8/8 lifetimes, one-sided p = 0.0039 (historical-rule diagnostic OCM_LIFETIME_RESIDUAL_SUPPORTED); all eight phase-G repairs succeed under the exact-predecessor check with exact rollback; kill gates 0. Under the merged regime the run is labelled CANNOT_CHECK_CURRENT_SCIENTIFIC_PROMOTION (engineering re-evaluation) and the cross-domain transfer family CANNOT_CHECK_MATCHED_CASES (6-vs-4 cells). Prospectively matched parent cells are implemented (phase_E matched_cells=True) for a V5 pre-registration under the merged regime.
